src/herder/HerderSCPDriver.cpp

```diff
@@ -469,6 +474,30 @@ HerderSCPDriver::timerCallbackWrapper(uint64_t slotIndex, int timerID,
     }
     else
     {
+        auto SCPTimingIt = mSCPExecutionTimes.find(slotIndex);
+        if (SCPTimingIt != mSCPExecutionTimes.end())
+        {
+            auto& SCPTiming = SCPTimingIt->second;
+            if (timerID == Slot::BALLOT_PROTOCOL_TIMER)
+            {
+                if (SCPTiming.mSCPProgressState ==
+                    SCPProgressState::FIRST_PREPARE)
+                {
+                    // Timeout happened in between first prepare and externalize
+                    mSCPMetrics.mPrepareTimeout.Mark();
```

Comment:
I think the metrics as measured are not going to be super useful: we should keep track of them in units of ledger (so maybe record the number of timeouts when we externalize using a histogram, the same way we track number of transactions/ledger)